### ISSUE TITLE
the loss is very easy to become a minus value 

### ISSUE THREAD

**Issue by Yusin2Chen** [SsnL/align_uniform#1]
Sorry to bother you, I trained my network using this two loss functions, but the loss is very easy to become a minus value, e.g. -0.00045.
Could you give me some help?
Thank you very much~

**Comment by ssnl** [comment]
Hello, the uniformity loss is not guaranteed to non-negative, so being negative is totally fine.

**Comment by Yusin2Chen** [comment]
> Hello, the uniformity loss is not guaranteed to non-negative, so being negative is totally fine.

Thanks ~

**Comment by ssnl** [comment]
We have updated the arXiv version to include a discussion on the range of Lunif. Please see Appendix A.1.1. in https://arxiv.org/pdf/2005.10242.pdf

In short, the optimal value is about `-2t` for large dimensionality.
